dll/win32/kernel32/client/thread.c

```diff
@@ -946,9 +951,14 @@ SetThreadUILanguage(IN LANGID LangId)
     }
 #endif
 
-    UNIMPLEMENTED;
+    PreviousLcid = Teb->CurrentLocale;
+    PreviousLangId = LANGIDFROMLCID(PreviousLcid);
+
+    NewLcid = MAKELCID(LangId, SORT_DEFAULT);
+
+    Teb->CurrentLocale = NewLcid;
```

Comment:
You should certainly call `SetThreadLocale(NewLcid);` instead. This will also allow to validate the new langid/lcid.

Reply:
And use `GetThreadLocale()` too above, I guess.